frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -400,8 +652,44 @@ export function ConfidenceMetricsSection({ artifacts }: ConfidenceMetricsSection
           ></IconWithTooltip>
         </h3>
       </div>
-      {/* TODO(zpChris): Introduce checkbox system that matches artifacts to curves. */}
-      <ROCCurve configs={rocCurveConfigs} />
+      <ROCCurve
+        configs={rocCurveConfigs}
+        colors={colors}
+        forceLegend={filter !== undefined}
+        disableAnimation={filter !== undefined}
```

Comment:
I force the legend to remain on the ROC Curve so that it does not disappear between selecting `1` and `2` artifacts, and I disable animation as that ended up creating confusing transitions during artifact selection.